entity-service-impl/src/main/java/org/hypertrace/entity/service/util/DocStoreConverter.java

```diff
@@ -106,6 +120,61 @@ private static boolean isPartOfAttributeMap(String fieldName) {
     return !EntityServiceConstants.ENTITY_CREATED_TIME.equalsIgnoreCase(fieldName);
   }
 
+//  private static void transform(AttributeValue attributeValue, Filter filter) throws IOException {
+//    if (ATTRIBUTES_LABELS_FIELD_NAME.equals(filter.getFieldName()) && filter.getOp() == Op.EQ) {
+//      transformToEqFilterWithValueListRhs(attributeValue, filter);
+//    } else if (ATTRIBUTES_LABELS_FIELD_NAME.equals(filter.getFieldName()) && filter.getOp() == Op.IN) {
+//      transformToOrFilterChainForStrArray(attributeValue, filter);
+//    } else {
+//      transform(attributeValue, filter, isPartOfAttributeMap(filter.getFieldName()));
+//    }
+//  }
+
+  private static void transformToOrFilterChainForStrArray(AttributeValue attributeValue, Filter filter) {
+    String fieldName = filter.getFieldName() + "." + "valueList" + "." + "values";
+
+    filter.setFieldName("");
+    filter.setOp(Op.OR);
+    filter.setChildFilters(
+        attributeValue.getValueList().getValuesList().stream()
+            .map(rhsAttributeValue -> {
+              Filter f = new Filter();
+              f.setFieldName(fieldName);
+              f.setOp(Op.EQ);
+
+              JsonNode mapNode = null;
+              try {
+                mapNode = OBJECT_MAPPER
+                    .readTree(JSONFORMAT_PRINTER.print(rhsAttributeValue));
+              } catch (JsonProcessingException | InvalidProtocolBufferException e) {
+                throw new RuntimeException(e);
+              }
+              Map map = OBJECT_MAPPER.convertValue(mapNode, Map.class);
+              f.setValue(map);
```

Comment:
I'm not sure if I understand this fully. Can you give an example please?

Reply:
What you want to have on the rhs is the same object format that's written for AttributeValue for label. So this:
`"attributes.labels.valueList.values": { "value": { "string": "foo" } }`. The JSON written down into mongo for the labels is `{'"valueList": {"values": [{ "value": { "string": "foo" } }, { "value": { "string": "bar" } }]}}`